util/dasharo/tag-with-changelog

```diff
@@ -45,35 +40,38 @@ else
 		echo
 		exit 1
 	fi
+
+	# Create a temporary lightweight tag, exit if it exists already
+	echo "Creating temporary $NEW_GIT_VERSION tag"
+	git tag $NEW_GIT_VERSION || exit 1
+
+	# Try to verify that the repo is clean before losing state.
+	if ! git diff-index --quiet --cached HEAD 2>/dev/null || \
+		[ "$(git diff $NEW_GIT_VERSION --shortstat 2>/dev/null | tail -n1)" != "" ]; then
```

Comment:
Just
```suggestion
	if ! git diff --quiet HEAD; then
```
instead of that upstream horror.

Reply:
It was added as part of https://review.coreboot.org/c/coreboot/+/62440, I suspect it was added to `Handle symbolic links to the git repo`. We probably don't have to deal with this corner case, though, it may be required for some specific downstream build system.

Reply:
https://github.com/Dasharo/coreboot/compare/74d14b8e589dc17fe4381940f7df82be6c8ce81b..4c3f9dc7da460387649522b54bcb68cf2f6a9f31